src/app/shared/element-child-data-classes-child-elements-list/flattened-data-classes-list.component.html

```diff
@@ -0,0 +1,339 @@
+          ></span>
+        </div>
+      </div>
+   
+    </ng-container>
+    <ng-container matColumnDef="element">
+      <th
+      mat-header-cell
+      *matHeaderCellDef
+      style="max-width: 35%; width: 35%"
+      columnName="element"
+      scope="col"
+    >
+    Data Element
+    </th>
+    <td
+      mat-cell
+      *matCellDef="let record"
+      style="overflow-wrap: break-word; word-break: break-word"
+    > 
+      <div>
+        <mdm-element-link
+        [element]="record"
+        [parentDataClass]="{id:record.dataClass}"
+        [parentDataModel]="{id:record.dataModel}"
+      ></mdm-element-link>
+        <div *ngIf="record.imported">
+          <span
+            class="badge badge-success item-type-imported"
+            title="Reference: {{
+              record.breadcrumbs | joinArray: ' / ':'label'
+            }} / {{ record.label }}"
+            >External Reference &nbsp;<span class="fas fa-info-circle"></span
+          ></span>
+        </div>
+      </div>
+      
+   
+    </ng-container>
+
+    <ng-container matColumnDef="description">
+      <th
+        mat-header-cell
+        *matHeaderCellDef
+        style="max-width: 50%; width: 50%"
+        columnName="description"
+        scope="col"
+      >
+        Description
+      </th>
+      <td mat-cell *matCellDef="let record">
+        <div
+          *ngIf="record.description && record.description.length > 0"
+          style="margin-bottom: 10px"
+        >
+          <mdm-more-description
+            [description]="record.description"
+          ></mdm-more-description>
+        </div>
+        <div>
+          <mdm-all-links-in-paged-list
+            [parent]="record"
+          ></mdm-all-links-in-paged-list>
+        </div>
+      </td>
+    </ng-container>
+    <ng-container matColumnDef="actions">
+      <td
+        mat-header-cell
+        *matHeaderCellDef
+        columnName="actions"
+        style="width: 5%; max-width: 5%; text-align: center"
+      ></td>
+      <td
+        mat-cell
+        *matCellDef="let record"
+        style="width: 5%; max-width: 5%; text-align: center"
+      >
+        <button
+          mat-icon-button
+          color="warn"
+          type="button"
+          (click)="toggleDelete(record)"
+          class="warning"
+          aria-label="Delete"
+          matTooltip="Delete"
+        >
+          <span class="far fa-trash-alt"></span>
+        </button>
+      </td>
+    </ng-container>
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr
+      mat-row
+      *matRowDef="let row; columns: displayedColumns"
+      cdkDrag
+      [cdkDragData]="row"
+      [ngClass]="{ 'mdm--active-row ': row.checked }"
+    ></tr>
+  </table>
+</div>
+<div
+  class="bordered is-hidden pxy-2"
+  [ngClass]="{ block: !totalItemCount && !isLoadingResults }"
+>
+  <p class="marginless">
+    There are no Data Classes in this model. You can add one by clicking the
```

Comment:
Confirm if this is the correct message to display when you have an empty model - from the perspective of the end user, does this read right?